mirar/pipelines/winter/blocks.py

```diff
@@ -661,48 +650,42 @@
     ),
 ]
 
-name_candidates = (
-    [
-        # Check if the source is already in the source table
-        SingleSpatialCrossmatchSource(
-            db_table=Source,
-            db_output_columns=["sourceid", SOURCE_NAME_KEY],
-            crossmatch_radius_arcsec=2.0,
-            ra_field_name="average_ra",
-            dec_field_name="average_dec",
-        ),
-        # Assign names to the new sources
-        CandidateNamer(
-            db_table=Source,
-            base_name=SOURCE_PREFIX,
-            name_start=NAME_START,
-            db_name_field=SOURCE_NAME_KEY,
-        ),
-        # Add the new sources to the source table
-        CustomSourceTableModifier(modifier_function=winter_new_source_updater),
-        DatabaseSourceInserter(
-            db_table=Source,
-            duplicate_protocol="ignore",
-        ),
-        # Get all candidates associated with source
-    ]
-    + select_history
-    + [
-        # Update average ra and dec for source
-        CustomSourceTableModifier(modifier_function=winter_source_entry_updater),
-        # Update sources in the source table
-        DatabaseSourceInserter(
-            db_table=Source,
-            duplicate_protocol="replace",
-        ),
-        # Add candidates in the candidate table
-        DatabaseSourceInserter(
-            db_table=Candidate,
-            duplicate_protocol="fail",
-        ),
-        SourceWriter(output_dir_name="preavro"),
-    ]
-)
+name_candidates = [
```

Comment:
Does this scheme work only if the SourceBatches are sorted ascending by time? Is that explicitly being done somewhere?

Reply:
Essentially the "history" is now "every detected candidate at this position". It would include later candidates, if e.g reprocessing old data. As a consequence, it makes no difference what order source batches are processed/ordered.

Reply:
OK

Reply:
I should also point out that the history-includes-later-detections behaviour was the case already. This PR basically just adds compatibility for history from images being processed at the same time.